elephant/spectral.py

```diff
@@ -523,6 +529,543 @@ def multitaper_psd(signal, n_segments=1, len_segment=None,
     return freqs, psd
 
 
+def multitaper_cross_spectrum(signals, fs=1., nw=4, num_tapers=None,
+                              peak_resolution=None, return_onesided=True,
+                              attach_units=True):
+    """
+    Estimates the cross spectrum of a given 'neo.AnalogSignal' using the
+    Multitaper method.
+
+    The cross spectrum is obtained through the following steps
+
+    1. obtain approximately independent estimates of the spectrum for each
+       signal by multiplying the segment with tapering functions (obtained as
+       the discrete prolate spheroidal functions, also known as slepian
+       functions) and calculating the cross spectrum for the tapered signals.
+       The number of tapering functions (and hence the number of the estimates)
+       is specified by the parameter `num_tapers`.
+
+    2. average the approximately independent estimates of each signal to
+       decrease overall variance of the estimates
+
+    Parameters
+    ----------
+    signal : neo.AnalogSignal or pq.Quantity or np.ndarray
+        Time series data of which PSD is estimated. When `signal` is
+        np.ndarray, it needs to be passed as a 2-dimensional array of shape
+        (n_channels, n_samples), and the data sampling frequency should be
+        given through the keyword argument `fs`.
+    fs : float, optional
+        Specifies the sampling frequency of the input time series
+        Default: 1.0.
+    frequency_resolution : pq.Quantity or float, optional
+        Desired frequency resolution of the obtained PSD estimate in terms of
+        the interval between adjacent frequency bins. When given as a `float`,
+        it is taken as frequency in Hz.
+        If None, it will be determined from other parameters.
+        Default: None.
+    nw : float, optional
+        Time-halfbandwidth product. This parameter can be used to determine the
+        number of tapers following the equation:
+            num_tapers = 2*nw - 1
+        It can be determined by multplying the duration of the signal with the
+        desired half-peak resolution frequency:
+            n_samples/fs * peak_resolution/2.
+        Default: 4.0.
+    num_tapers : int, optional
+        Number of tapers used in step 2 (see above) to obtain estimate of PSD.
+        By default, [2*nw] - 1 is chosen.
+        Default: None.
+    peak_resolution : pq.Quantity float, optional
+        Quantity in Hz determining the number of tapers used for analysis.
+        Fine peak resolution --> low numerical value --> low number of tapers
+        High peak resolution --> high numerical value --> high number of tapers
+        When given as a `float`, it is taken as frequency in Hz.
+        Default: None.
```

Comment:
When a parameter takes `None`, the expected behavior should be informed. If there is a complex relationship between the parameters and how the `None`s are treated, so that the descriptive text of each parameter would be too long or with several redundancies, then the description can be put in an explanation note below. In this case, this text would have the reference to that note (e.g., `see Notes [1]`).

Reply:
Addressed in https://github.com/NeuralEnsemble/elephant/pull/525/commits/5ddd042c88783fdc4203819a36146c746510599d. Could you please have a look at the changes made and tell us if this is what you meant?